lib/common/xml.c

```diff
@@ -1350,16 +1367,17 @@ crm_xml_escape(const char *text)
         return NULL;
     }
 
-    length = 1 + strlen(text);
+    length = strlen(text);
     copy = strdup(text);
     CRM_ASSERT(copy != NULL);
-    for (size_t index = 0; index < length; index++) {
+    for (size_t index = 0; index <= length; index++) {
```

Comment:
This should still be `< length`. The test was off by one in the first place. The new `pcmk__xml_escape()` uses `< length`.